internal/cmd/users.go

```diff
@@ -225,51 +227,68 @@ func updateUser(cli *CLI, name string) error {
 		return err
 	}
 
-	req := &api.UpdateUserRequest{}
+	config, err := currentHostConfig()
+	if err != nil {
+		return err
+	}
 
 	isSelf, err := isUserSelf(name)
 	if err != nil {
 		return err
 	}
 
 	if isSelf {
-		config, err := currentHostConfig()
-		if err != nil {
+		req := &api.UpdateUserRequest{}
+		if req.ID, err = config.UserID.ID(); err != nil {
 			return err
 		}
 
-		if req.ID, err = config.PolymorphicID.ID(); err != nil {
+		fmt.Fprintf(cli.Stderr, "  Enter a new password (min. length 8):\n")
+		req.Password, err = promptSetPassword(cli, "")
+		if err != nil {
 			return err
 		}
-	} else {
-		user, err := getUserByName(client, name)
-		if err != nil {
-			if errors.Is(err, ErrUserNotFound) {
-				return Error{Message: fmt.Sprintf("No user named %q in local provider; only local users can be edited", name)}
-			}
+
+		if _, err := client.UpdateUser(req); err != nil {
 			return err
 		}
 
-		req.ID = user.ID
+		cli.Output("  Updated password")
+
+		return nil
 	}
 
-	fmt.Fprintf(cli.Stderr, "  Enter a new password (min. length 8):\n")
-	req.Password, err = promptSetPassword(cli, "")
+	ok, err := hasAccessToChangePasswordsForOtherUsers(client, config)
 	if err != nil {
 		return err
 	}
+	if !ok {
+		return Error{Message: "No permission to change password for user " + name}
+	}
 
-	logging.S.Debugf("call server: update user %s", req.ID)
-	if _, err := client.UpdateUser(req); err != nil {
+	user, err := getUserByName(client, name)
+	if err != nil {
+		if errors.Is(err, ErrUserNotFound) {
+			logging.S.Debugf("user not found: %s", err)
```

Comment:
This debug log seems repetitive of the actual error

Reply:
we lose the actual root error; thought it might be helpful